vaadin-crud-flow-parent/vaadin-crud-flow-integration-tests/src/main/java/com/vaadin/flow/component/crud/tests/PersonCrudDataProvider.java

```diff
@@ -53,9 +53,10 @@ protected Stream<Person> fetchFromBackEnd(Query<Person, CrudFilter> query) {
 
         Stream<Person> stream = getDatabaseCopy().stream();
 
-        if (query.getFilter().isPresent()) {
-            stream = stream.filter(predicate(query.getFilter().get()))
-                    .sorted(comparator(query.getFilter().get()));
+        var filter = query.getFilter().orElse(null);
+        if (filter != null) {
+            stream = stream.filter(predicate(filter))
+                    .sorted(comparator(filter));
```

Comment:
I can't understand why sonarcloud is complaining about this when one of the compliant solution option is to use `isPresent()`. 🤔

Reply:
I guess it's because `query.getFilter()` could potentially return a different value on each call.